src/java/org/apache/cassandra/index/sai/disk/vector/OnDiskOrdinalsMap.java

```diff
@@ -259,6 +270,26 @@ public int getOrdinalForRowId(int rowId) throws IOException
             return reader.readInt();
         }
 
+        @Override
+        public void forEachOrdinalInRange(int startRowId, int endRowId, OrdinalConsumer consumer) throws IOException
+        {
+            reader.seek(rowOrdinalOffset);
+            // sequential read without seeks should be fast, we sxpect OS to prefetch data from the disk
+            // binary search for starting offset of min rowid >= startRowId unlikely to be faster
+            for (long idx = 0; idx <= high; idx ++)
+            {
+                int rowId = reader.readInt();
+                if (rowId > endRowId)
+                    break;
+
+                int ordinal = reader.readInt();
```

Comment:
Do you know if there is a meaningful difference between `skipBytes(4)` and `readInt()`?

Reply:
I don't think skipBytes is applicable there (I still need rowId to stop if requested range is shorter than the written one).
Other than that, skipBytes(4) will not be measurably faster form a quick peek into the code

Reply:
Makes sense. It was only going to help us when the `rowId < startRowId`